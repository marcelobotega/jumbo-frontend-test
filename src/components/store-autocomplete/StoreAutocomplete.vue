<template>
  <div class="store-autocomplete-wrapper form-group">
    <div class="form-control-icon">
      <input
        placeholder="Search stores or cities"
        type="text"
        name="search"
        autocomplete="off"
        class="form-control"
        v-model="inputValue"
        @input="doSearchDebounced"
        @keydown.down.prevent="onArrowDown"
        @keydown.up.prevent="onArrowUp"
        @keydown.enter.prevent="onEnter"
      />
      <div class="form-icon">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 20 20"
          fill="currentColor"
        >
          <path
            fill-rule="evenodd"
            d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
            clip-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div v-if="inputValue && typed" class="store-autocomplete-panel">
      <p class="text-xs text-gray-500">
        Use the up and down arrow keys to move the selection
      </p>
      <h4>Stores</h4>
      <div class="ml-2">
        <div v-if="stores.length === 0">
          <p class="text-sm font-semibold mt-1">
            Sorry we couldn't find any stores with your search term.
          </p>
        </div>
        <div v-else>
          <a
            class="block hover:bg-red-100 focus:bg-red-100"
            href="#"
            tabindex="0"
            v-for="(store, i) in stores"
            :key="i"
            :class="{ 'bg-red-100': i === listPosition }"
            @click.prevent="itemSelected(store)"
          >
            {{ store }}
          </a>
          <p v-if="stores.length === 10" class="text-sm font-semibold mt-1">
            Search limited to 10 stores. Refine search terms to make your search
            results more precise
          </p>
        </div>
      </div>

      <h4 class="mt-1">Cities</h4>
      <div class="ml-2">
        <div v-if="cities.length === 0">
          <p class="text-sm font-semibold mt-1">
            Sorry we couldn't find any city with your search term.
          </p>
        </div>
        <div v-else>
          <a
            class="block hover:bg-green-100 focus:bg-green-100"
            href="#"
            tabindex="0"
            v-for="(city, i) in cities"
            :key="i + stores.length"
            :class="{ 'bg-green-100': i + stores.length === listPosition }"
            @click.prevent="itemSelected(city)"
          >
            {{ city }}
          </a>
          <p v-if="cities.length === 10" class="text-sm font-semibold mt-1">
            Search limited to 10 cities. Refine search terms to make your search
            results more precise
          </p>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Emit, Vue } from "vue-property-decorator";
import _debounce from "lodash/debounce";
import storeService from "@/services/store.service";
import citiesService from "@/services/cities.service";
import { IStore } from "@/domain/store";

@Component
export default class StoreAutocomplete extends Vue {
  doSearchDebounced = _debounce(this.doSearch, 300);
  stores: string[] = [];
  cities: string[] = [];
  inputValue = "";
  listPosition = 0;
  typed = false;

  private resultsLimit = 10;
  private results: string[] = [];

  mounted(): void {
    document.addEventListener("click", this.handleClickOutside);
  }

  destroyed(): void {
    document.removeEventListener("click", this.handleClickOutside);
  }

  handleClickOutside(event: Event): void {
    if (event && !this.$el.contains(event.target as HTMLDivElement)) {
      this.resetState();
    }
  }

  doSearch(e: Event): void {
    this.resetState();
    const search = (e.target as HTMLInputElement).value.trim();
    if (search) {
      this.typed = true;

      this.stores = storeService
        .getStoresOrdered(true, "addressName", search)
        .map((store: IStore) => store.addressName)
        .slice(0, this.resultsLimit);

      this.cities = citiesService
        .getCitiesOrdered(true, search)
        .slice(0, this.resultsLimit);

      this.results = [...this.stores, ...this.cities];
    } else {
      this.itemSelected("");
    }
  }

  onArrowDown(): void {
    if (this.listPosition < this.results.length - 1) {
      this.listPosition++;
    } else {
      this.listPosition = 0;
    }
  }

  onArrowUp(): void {
    if (this.listPosition > 0) {
      this.listPosition--;
    } else {
      this.listPosition = this.results.length - 1;
    }
  }

  onEnter(): void {
    const autoCompleteItem = this.results[this.listPosition];

    if (autoCompleteItem) {
      this.itemSelected(this.results[this.listPosition]);
    } else {
      this.itemSelected("");
    }

    this.resetState();
  }

  @Emit()
  itemSelected(item: string): string {
    this.resetState();
    this.inputValue = item;
    return item;
  }

  private resetState(): void {
    this.typed = false;
    this.listPosition = 0;
    this.stores = [];
    this.cities = [];
    this.results = [];
  }
}
</script>
<style scoped lang="scss">
.store-autocomplete-wrapper {
  @apply md:w-1/2
    lg:w-1/4
    sm:w-full
    relative;

  .store-autocomplete-panel {
    @apply bg-white
      z-50
    absolute
    px-4
    py-2
    rounded
    shadow-lg
    border border-gray-400;
  }
}
</style>
