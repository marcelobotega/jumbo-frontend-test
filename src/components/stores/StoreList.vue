<template>
  <div class="stores-list-wrapper content-section">
    <h1>Stores</h1>
    <div class="stores-list-content mt-4">
      <div class="stores-list-search mb-3">
        <div class="md:w-1/2 lg:w-1/4 sm:w-full form-group">
          <label for="search" class="form-label">Search:</label>
          <div class="form-control-icon">
            <input
              type="text"
              name="search"
              id="search"
              autocomplete="off"
              class="form-control"
              @input="doSearchDebounced"
            />
            <div class="form-icon">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 20 20"
                fill="currentColor"
              >
                <path
                  fill-rule="evenodd"
                  d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                  clip-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <hr />
      <StoreCard
        v-for="store in stores"
        :key="store.uuid"
        class="mt-3"
        :store="store"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { IStore } from "@/domain/store";
import storesService from "@/services/store.service";
import _debounce from "lodash/debounce";
import StoreCard from "@/components/store-card/StoreCard.vue";

@Component({
  components: { StoreCard },
})
export default class StoreList extends Vue {
  stores: IStore[] = storesService.getStoresOrdered(true, "addressName");

  doSearchDebounced = _debounce(this.doSearch, 500);

  doSearch(e: Event): void {
    const searchValue = (e.target as HTMLInputElement).value;

    this.stores = storesService.getStoresOrdered(
      true,
      "addressName",
      searchValue
    );
  }
}
</script>
