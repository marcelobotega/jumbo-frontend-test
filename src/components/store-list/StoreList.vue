<template>
  <div class="stores-list-wrapper content-section">
    <h1>Stores</h1>
    <div class="stores-list-content mt-4">
      <div class="stores-list-search mb-3">
        <StoreAutocomplete @item-selected="doSearch" />
      </div>
      <hr />
      <div v-if="stores.length === 0" class="mt-3">
        <p class="text-sm font-semibold mt-1">
          Sorry we couldn't find any stores with your search term.
        </p>
      </div>
      <div v-else>
        <StoreCard
          v-for="store in stores"
          :key="store.uuid"
          class="mt-3"
          :store="store"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { IStore } from "@/domain/store";
import storeService from "@/services/store.service";
import StoreCard from "@/components/store-card/StoreCard.vue";
import StoreAutocomplete from "@/components/store-autocomplete/StoreAutocomplete.vue";

@Component({
  components: { StoreAutocomplete, StoreCard },
})
export default class StoreList extends Vue {
  stores: IStore[] = storeService.getStoresOrdered(true, "addressName");

  doSearch(searchValue: string): void {
    this.stores = storeService.getStoresOrdered(
      true,
      "addressName",
      searchValue
    );
  }
}
</script>
