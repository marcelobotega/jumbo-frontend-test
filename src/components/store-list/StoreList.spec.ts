import { mount, Wrapper } from "@vue/test-utils";
import StoreList from "@/components/store-list/StoreList.vue";
import storesService from "@/services/store.service";
import StoreAutocomplete from "@/components/store-autocomplete/StoreAutocomplete.vue";

const storeOneMock = {
  city: "St. Nicolaasga",
  postalCode: "8521 KD",
  street: "Molewei",
  street2: "1",
  street3: "",
  addressName: "Jumbo St. Nicolaasga Jan Altenburg",
  uuid: "n5MKYx4XLxAAAAFIG.EYwKxJ",
  latitude: "52.923021",
  longitude: "5.740684",
  complexNumber: "33250",
  locationType: "SupermarktPuP",
  collectionPoint: true,
  sundayOpen: true,
  todayOpen: "08:00",
  todayClose: "20:00",
  sapStoreID: "3420",
  showWarningMessage: true,
};

const storeTwoMock = {
  city: "St. Oedenrode",
  postalCode: "5492 BA",
  street: "Hertog Hendrikstraat",
  street2: "1",
  street3: "",
  addressName: "Jumbo St. Oedenrode Hertog Hendrikstraat",
  uuid: "JygKYx4XSVYAAAFlg8UV9qHu",
  latitude: "51.561974",
  longitude: "5.460663",
  complexNumber: "32360",
  locationType: "SupermarktPuP",
  collectionPoint: true,
  sundayOpen: true,
  todayOpen: "08:00",
  todayClose: "21:00",
  sapStoreID: "5127",
  showWarningMessage: true,
};

describe("StoreList.vue", () => {
  let wrapper: Wrapper<StoreList>;
  const getStoresOrderedSpy = jest.spyOn(storesService, "getStoresOrdered");

  beforeEach(() => {
    getStoresOrderedSpy.mockReturnValueOnce([storeOneMock, storeTwoMock]);
    wrapper = mount(StoreList);
  });

  it("should create", () => {
    expect(getStoresOrderedSpy).toHaveBeenCalledTimes(1);
    expect(getStoresOrderedSpy).toHaveBeenCalledWith(true, "addressName");
    expect(wrapper).toMatchSnapshot();
  });

  it("should create test", async () => {
    jest.resetAllMocks();
    getStoresOrderedSpy.mockReturnValueOnce([storeTwoMock]);

    wrapper
      .findComponent(StoreAutocomplete)
      .vm.$emit("item-selected", "Jumbo St. Oedenrode Hertog Hendrikstraat");

    await wrapper.vm.$nextTick();
    expect(getStoresOrderedSpy).toHaveBeenCalledTimes(1);
    expect(getStoresOrderedSpy).toHaveBeenCalledWith(
      true,
      "addressName",
      "Jumbo St. Oedenrode Hertog Hendrikstraat"
    );
    expect(wrapper).toMatchSnapshot();
  });
});
