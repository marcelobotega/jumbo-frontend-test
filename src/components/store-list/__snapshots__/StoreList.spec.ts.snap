// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreList.vue should create 1`] = `
<div class="stores-list-wrapper content-section">
  <h1>Stores</h1>
  <div class="stores-list-content mt-4">
    <div class="stores-list-search mb-3">
      <div class="store-autocomplete-wrapper form-group">
        <div class="form-control-icon"><input placeholder="Search stores or cities" type="text" name="search" autocomplete="off" class="form-control">
          <div class="form-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
              <path fill-rule="evenodd" d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z" clip-rule="evenodd"></path>
            </svg></div>
        </div>
        <!---->
      </div>
    </div>
    <hr>
    <div><a href="https://www.google.com/maps/search/?api=1&amp;query=Jumbo St. Nicolaasga Jan Altenburg" target="_blank" class="store-card card mt-3">
        <div class="card-content">
          <h4>Jumbo St. Nicolaasga Jan Altenburg</h4>
          <div class="extras flex flex-wrap">
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="map-marker-alt" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"></path>
              </svg>
              Molewei 1
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="city" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M616 192H480V24c0-13.26-10.74-24-24-24H312c-13.26 0-24 10.74-24 24v72h-64V16c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v80h-64V16c0-8.84-7.16-16-16-16H80c-8.84 0-16 7.16-16 16v80H24c-13.26 0-24 10.74-24 24v360c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V216c0-13.26-10.75-24-24-24zM128 404c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm128 192c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12V76c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 288c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40z"></path>
              </svg>

              8521 KD St. Nicolaasga
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="clock" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path>
              </svg>
              Open 08:00 to 20:00<span>, Opens sunday</span></p>
          </div>
        </div>
      </a><a href="https://www.google.com/maps/search/?api=1&amp;query=Jumbo St. Oedenrode Hertog Hendrikstraat" target="_blank" class="store-card card mt-3">
        <div class="card-content">
          <h4>Jumbo St. Oedenrode Hertog Hendrikstraat</h4>
          <div class="extras flex flex-wrap">
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="map-marker-alt" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"></path>
              </svg>
              Hertog Hendrikstraat 1
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="city" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M616 192H480V24c0-13.26-10.74-24-24-24H312c-13.26 0-24 10.74-24 24v72h-64V16c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v80h-64V16c0-8.84-7.16-16-16-16H80c-8.84 0-16 7.16-16 16v80H24c-13.26 0-24 10.74-24 24v360c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V216c0-13.26-10.75-24-24-24zM128 404c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm128 192c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12V76c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 288c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40z"></path>
              </svg>

              5492 BA St. Oedenrode
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="clock" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path>
              </svg>
              Open 08:00 to 21:00<span>, Opens sunday</span></p>
          </div>
        </div>
      </a></div>
  </div>
</div>
`;

exports[`StoreList.vue should create test 1`] = `
<div class="stores-list-wrapper content-section">
  <h1>Stores</h1>
  <div class="stores-list-content mt-4">
    <div class="stores-list-search mb-3">
      <div class="store-autocomplete-wrapper form-group">
        <div class="form-control-icon"><input placeholder="Search stores or cities" type="text" name="search" autocomplete="off" class="form-control">
          <div class="form-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
              <path fill-rule="evenodd" d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z" clip-rule="evenodd"></path>
            </svg></div>
        </div>
        <!---->
      </div>
    </div>
    <hr>
    <div><a href="https://www.google.com/maps/search/?api=1&amp;query=Jumbo St. Oedenrode Hertog Hendrikstraat" target="_blank" class="store-card card mt-3">
        <div class="card-content">
          <h4>Jumbo St. Oedenrode Hertog Hendrikstraat</h4>
          <div class="extras flex flex-wrap">
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="map-marker-alt" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"></path>
              </svg>
              Hertog Hendrikstraat 1
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="city" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M616 192H480V24c0-13.26-10.74-24-24-24H312c-13.26 0-24 10.74-24 24v72h-64V16c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v80h-64V16c0-8.84-7.16-16-16-16H80c-8.84 0-16 7.16-16 16v80H24c-13.26 0-24 10.74-24 24v360c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V216c0-13.26-10.75-24-24-24zM128 404c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm128 192c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12V76c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 288c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40z"></path>
              </svg>

              5492 BA St. Oedenrode
            </p>
            <p class="text-sm mt-1 mr-4 inline text-gray-700"><svg aria-hidden="true" focusable="false" data-icon="clock" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="w-4 h-4 text-gray-400 inline mr-1 align-text-bottom">
                <path fill="currentColor" d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path>
              </svg>
              Open 08:00 to 21:00<span>, Opens sunday</span></p>
          </div>
        </div>
      </a></div>
  </div>
</div>
`;
